packages/orchestrator-ui-components/src/components/WFOSteps/WFOStepList/WFOStepList.tsx

```diff
@@ -0,0 +1,129 @@
+
+    const {
+        stepSpacerStyle,
+        stepListHeaderStyle,
+        stepListContentStyle,
+        stepListContentBoldTextStyle,
+        stepListContentAnchorStyle,
+    } = getStyles(theme);
+
+    return (
+        <>
+            <EuiFlexGroup css={stepListHeaderStyle}>
+                <EuiFlexGroup css={stepListContentStyle}>
+                    <EuiText css={stepListContentBoldTextStyle}>
+                        {t('steps')}
+                    </EuiText>
+                    <EuiText
+                        css={stepListContentAnchorStyle}
+                        onClick={openAllDetails}
+                    >
+                        {t('expandAll')}
+                    </EuiText>
+                </EuiFlexGroup>
+                <EuiFlexGroup
+                    justifyContent="flexEnd"
+                    direction="row"
+                    css={{ flexGrow: 0 }}
+                    gutterSize="s"
+                >
+                    <EuiButton
+                        onClick={(
+                            e: React.MouseEvent<
+                                HTMLButtonElement | HTMLElement,
+                                MouseEvent
+                            >,
+                        ) => {
+                            e.preventDefault();
+                            alert('TODO: Implement Show delta');
+                        }}
+                        iconSide="right"
+                        size="s"
+                        iconType="visVega"
+                    >
+                        {t('showDelta')}
+                    </EuiButton>
+                    <EuiButton
+                        onClick={(
+                            e: React.MouseEvent<
+                                HTMLButtonElement | HTMLElement,
+                                MouseEvent
+                            >,
+                        ) => {
+                            e.preventDefault();
+                            alert('TODO: Implement View options');
+                        }}
+                        iconType="eye"
+                        iconSide="right"
+                        size="s"
+                    >
+                        {t('viewOptions')}
+                    </EuiButton>
+                </EuiFlexGroup>
+            </EuiFlexGroup>
+            <>
+                {steps.map((step, index) => {
+                    const stepComponent = (
+                        <div key={`step-${index}`}>
+                            {index !== 0 && <div css={stepSpacerStyle} />}
+                            <WFOStep
+                                stepDetailIsOpen={
+                                    stepDetailStates.get(index) || false
+                                }
+                                toggleStepDetailIsOpen={toggleStepDetailIsOpen}
+                                step={step}
+                                stepIndex={index}
+                                startedAt={startedAt}
```

Comment:
this could be `index > 0 ? step.executed : startedAt`

Reply:
This doesn't do the same thing. The startedAt time is updated for every step so it get's the executed time from the previous step to be able to calculate the duration of the step

Reply:
Ok, but still reassigning a value from props is tricky. Is it possible to use useState here?

Reply:
I think useState would cause a rerender everytime it's set. I have copied the startvalue to another variable so it doesn't change the prop anymore